awx/main/tests/functional/conftest.py

```diff
@@ -122,6 +122,22 @@ def __get__(self, obj, obj_type):
     mocked.start()
 
 
+@pytest.fixture
+def run_computed_fields_right_away(request):
+
+    def run_me(inventory_id, should_update_hosts=True):
+        i = Inventory.objects.get(id=inventory_id)
+        i.update_computed_fields(update_hosts=should_update_hosts)
+
+    mocked = mock.patch(
+        'awx.main.signals.update_inventory_computed_fields.delay',
+        new=run_me
+    )
+    mocked.start()
+
+    request.addfinalizer(mocked.stop)
```

Comment:
I force pushed a modification to add this finalizer.

This is kind of an absurdly basic use case for me to be in this situation - but I can't find any other examples of tests doing this. I really want this to be a general fixture, and the very simple thing I want is "if this fixture is used, then mock stuff for the duration of the test".

As far as I've been able to research, this seems to be the most accepted way of doing it.

Reply:
is `mocked.stop` a callable that takes no args, e.g. `()` ?

Reply:
if so I think you are good. You can watch what is happening by running a test that uses this fixture and add `--setup-show` to your pytest args. This will show you all setup and teardown steps

Reply:
yes, same as `start`

Reply:
That is cool

https://gist.github.com/AlanCoding/3ada634e7327942f5c31c5438e9f0f3c

I guess nothing really jumps out at me. Looks like it's working.